### DISCUSSION TITLE
terminating container during PULLING

### DISCUSSION THREAD

**Discussion by youngjun0627** [lablup/contribution-academy-2021#34] [Q&A] [answered]
### Describe the problem
PULLING 상태인 container를 강제종료 및 삭제를 시도하였으나 
`BackendAPIError: 403 Forbidden`
에러가 뜨면서 커널이 종료되지 않습니다.

### To Reproduce
개발환경을 설치한 후 'hello world' 까지 진행하는데 이상이 없습니다.
그러나 mnist demo를 진행하기 위하여 image를 갖고 올 때 [백엔드닷에이아이(Backend.AI)로 시작해 보는 AI 오픈소스 소개 및 실습 - 04](https://www.youtube.com/watch?v=steEbO0SG5E&list=PLYkiFpaI5DIJNwFbUCvtwlDg3R1Q9A2D4&index=8)  과 달리 
`backend.ai admin images` 를 통해 찾은 이미지를 이용하여 다음과 같이 aliasing을 해주었습니다.
`backend.ai admin alias-image tf2 cf.backend.ai/stable/python-tensorflow:2.4-py38-cuda11.1`

그 후 다음 명령어를 실행하였습니다.
`backend.ai run --rm -r cpu=2 -r mem=1G -c "import tensorflow as tf;print(tf.__version__)' tf2`
그러나 이 과정에서 아무 반응이 없어 강제종료를 하였고, 생성된 컨테이너를 지우기 위해 `backend.ai rm <해당 컨테이너id>`를 입력하였으나 `Cannot destroy kernels in pulling status`에러가 발생하며 작업이 취소됩니다. --forced 옵션도 추가하였으나 같은 에러가 발생합니다.

### Expected behavior
컨테이너가 삭제되는 것을 기대하였습니다.

### Screenshots

1.  `backend.ai run --rm -r cpu=2 -r mem=1G -c "import tensorflow as tf;print(tf.__version__)' tf2`

![image](https://user-images.githubusercontent.com/57058726/128862136-b067bb08-b5c7-45fe-96f1-c8513954f7c4.png)

2.  `backend.ai rm <해당 컨테이너id>`
![image](https://user-images.githubusercontent.com/57058726/128862071-c7c015a7-fe2f-4a0c-9a66-3bbbcdf8ae50.png)

### Environmen Description
OS: Ubuntu 20.04
Python version: 3.9.5
Current Shell: bash

**Comment by lizable** [comment, marked as answer]
@youngjun0627 
For now, the policy of termination only allows in these status:
- RUNNING
- PENDING

available when forcefully terminated:
- SCHEDULING
- PREPARING
- TERMINATING

So, sessions in PULLING status cannot be destroyed until the image pulling operation in the agent has done. To see more detailed information, please check [this link](https://github.com/lablup/backend.ai/issues/163).

**Reply by lizable** [reply]
TL;DR, You can't terminate the session in pulling status until its operation ends.

**Reply by youngjun0627** [reply]
I've reinstalled...

**Reply by youngjun0627** [reply]
@lizable 
I find how to terminate the session in pulling status!
1. restart the session. `backend.ai restart <the session name>`
2. command `backend.ai ps`, and check if the status of that is restarting.
3. command `backend.ai rm <the session name>`
but, it is not normal termination...

**Reply by lizable** [reply]
Yes, you can remove it forcefully, but It will bring resource recalculation. Eventually we should provide a progress of pulling process, so that the user could understand it takes time.
